### PR TITLE
RFR: Annotations integration tests

### PR DESCRIPTION
**Pull request by tilogaat**
Continuation of PR #484 

Adding a lot of integration tests for the annotations API.

### CODE REVIEW

File: blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/inputs/handlers/HttpAnnotationsEndToEndTest.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        HttpGet get = new HttpGet(getAnnotationsQueryURI());
+        response = client.execute(get);
+
+        String responseString = EntityUtils.toString(response.getEntity());
+        Assert.assertEquals(200, response.getStatusLine().getStatusCode());
+        Assert.assertFalse(responseString.equals("[]"));
+        Assert.assertTrue(responseString.contains("deployment 0"));
+    }
+
+    @AfterClass
```
Comment (chinmay-gupte) [inline blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/inputs/handlers/HttpAnnotationsEndToEndTest.java]: NIT: I usually prefer the teardown method at the bottom of the class.

File: blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/inputs/handlers/HttpHandlerIntegrationTest.java [hunk 2]
```diff
@@ -220,8 +336,48 @@ private URIBuilder getMetricsURIBuilder() throws URISyntaxException {
...
+            event.setWhen(Calendar.getInstance().getTimeInMillis());
+            event.setData("deploying prod");
+            event.setTags("deployment");
+
+            eventList.add(event.toMap());
+        }
+        eventsSearchIO.insert(tenant, eventList);
+    }
+
+    private HttpResponse postEvent(String requestBody, String tenantId) throws Exception {
```
Comment (chinmay-gupte) [inline blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/inputs/handlers/HttpHandlerIntegrationTest.java]: Is it possible to combine these helper methods with the helper methods in `HttpAnnotationsEndToEndTest`?

Comment (chinmay-gupte) [inline blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/inputs/handlers/HttpHandlerIntegrationTest.java]: They are exact copy of each other right?

Comment (tilogaat) [inline blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/inputs/handlers/HttpHandlerIntegrationTest.java]: @chinmay-gupte yes they are, can you suggest a good place such as a test utils class to put them in? Inheriting from IntegrationTestBase make the test really slow and they would need the dependency of Cassandra being run always.

File: blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/inputs/handlers/HttpAnnotationsEndToEndTest.java (hunk 1 above)
Comment (tilogaat) [inline blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/inputs/handlers/HttpAnnotationsEndToEndTest.java]: 9d01902

File: blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/inputs/handlers/HttpHandlerIntegrationTest.java (hunk 2 above)
Comment (tilogaat) [inline blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/inputs/handlers/HttpHandlerIntegrationTest.java]: Fixed 28ef5e7

File: blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/outputs/handlers/HttpAnnotationsIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        Assert.assertNotNull(responseString);
+        Assert.assertTrue(responseString.equals("[]"));
+
+        parameterMap.put(Event.tagsParameterName, "{0,1}");
+        get = new HttpGet(getAnnotationsQueryURI());
+        response = client.execute(get);
+        responseString = EntityUtils.toString(response.getEntity());
+        Assert.assertNotNull(responseString);
+        Assert.assertTrue(responseString.equals("[]"));
+
```
Comment (chinmay-gupte) [inline blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/outputs/handlers/HttpAnnotationsIntegrationTest.java]: NIT: extra new line.

File: blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/outputs/handlers/HttpAnnotationsIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,182 @@
+package com.rackspacecloud.blueflood.outputs.handlers;
```
Comment (chinmay-gupte) [inline blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/outputs/handlers/HttpAnnotationsIntegrationTest.java]: Missing license file.

**Comment by chinmay-gupte** [comment]
LGTM. +1 after fixing the nit and adding the missing license file. Also, can you squash the commits before merging?

**Pull request merged**